### PR TITLE
Bump BT v5 To Xcode 15.0.1 & Swift 5.9+

### PR DESCRIPTION
**Pull request by agedd**
### Summary of changes

Apple is requesting that apps submitted to the App Store Connect must be built with Xcode 15 [here](https://developer.apple.com/news/upcoming-requirements/?id=04292024a).

- This PR bumps BT iOS v5 to `Xcode 15.0+` & `Swift 5.9+`
- Updated `Cocoapods` to version `1.15.2`
- Updated `CHANGELOG` and `README` files

### Checklist

- [x] Added a changelog entry

- @agedd

### CODE REVIEW

**Comment by agedd** [comment]
buildit

**Comment by agedd** [comment]
hmm seems like CI is failing with this error: ` xcode-select: error: invalid developer directory '/Applications/Xcode_15.1.app'` but not sure how to resolve this yet (working fine locally) 😅

**Comment by scannillo** [comment]
> hmm seems like CI is failing with this error:  xcode-select: error: invalid developer directory '/Applications/Xcode_15.1.app' looking into how to resolve this but seems to be working fine locally 😅

You might need to bump the [MacOS version](https://github.com/actions/runner-images/tree/main/images/macos) to get access to the Xcode 15 image? Sometimes `macos-latest` doesn't properly pull in the latest

**Review by stechiu — APPROVED** [review]

**Review by jaxdesmarais — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -8,15 +8,15 @@ on:
...
     steps:
       - name: Check out repository
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
 
-      - name: Use Xcode 14.1
-        run: sudo xcode-select -switch /Applications/Xcode_14.1.app
+      - name: Use Xcode 15.1
+        run: sudo xcode-select -switch /Applications/Xcode_15.1.app
```
Comment (jaxdesmarais) [inline .github/workflows/release.yml:19]: We will want to use 15.0.1 to support Carthage merchants using Xcode 15.0 ([slack convo here](https://paypal.slack.com/archives/C03U9SEQYHK/p1711464753326179?thread_ts=1711387809.750689&cid=C03U9SEQYHK)). Up to you on if we want to update everything to 15.0.1 or just this portion. It looks like the macOS-14 runner should support this: https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md#xcode

Comment (agedd) [reply .github/workflows/release.yml]: thanks for letting me know @jaxdesmarais! updated [here](b27be70ff88fe05890851e2aeb4e242bc052eddf)

**Review by jaxdesmarais — COMMENTED** [review]
We've talked in a few other PRs about also bumping the Package.swift and podspec to Swift 5.9. I think we can likely do the same here (I've added a ticket next sprint to do the same on v6)

**Comment by agedd** [comment]
> We've talked in a few other PRs about also bumping the Package.swift and podspec to Swift 5.9. I think we can likely do the same here (I've added a ticket next sprint to do the same on v6)

sweet, addressed [here](2e635ab47aa964da48346bc56cbc3abc6f6c5b62)!

**Pull request merged**